stl/inc/xmemory

```diff
@@ -675,35 +686,36 @@ struct _Default_allocator_traits { // traits for std::allocator
     template <class _Other>
     using rebind_traits = allocator_traits<allocator<_Other>>;
 
-    _NODISCARD static __declspec(allocator) pointer allocate(_Alloc&, _CRT_GUARDOVERFLOW const size_type _Count) {
+    _NODISCARD static _CONSTEXPR20_DYNALLOC __declspec(allocator) pointer

```

Comment:
So this needs some love for MSVC, as only `std::allocator::allocate` is allowed during constant evalaution:

```cpp    
    _NODISCARD static _CONSTEXPR20_DYNALLOC __declspec(allocator) pointer
        allocate(_Alloc& _Al, _CRT_GUARDOVERFLOW const size_type _Count) {
#ifdef __cpp_lib_is_constant_evaluated
        if (_STD is_constant_evaluated()) {
            return _Al.allocate(_Count);
        } else
#endif // __cpp_lib_is_constant_evaluated
        {
            (void) _Al;
            return static_cast<pointer>(
                _Allocate<_New_alignof<value_type>>(_Get_size_of_n<sizeof(value_type)>(_Count)));
        }
    }
```

This also applies for the other functions below